test/session_tests/DocumentQueryTest.ts

```diff
@@ -8,220 +8,493 @@ import {RequestExecutor} from "../../src/Http/Request/RequestExecutor";
 
-    it('should query by double index joined by "AND" operator', async() => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product,
-        waitForNonStaleResults: true,
-        usingDefaultOperator: QueryOperators.AND
-      })
-      .whereEquals<string>('name', 'test107')
-      .whereEquals<number>('uid', 5).get();
+    it('should query with orderBy as long', async () => {
+      const results: Universal[] = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .orderBy<string>('order as long', 'ASC').get();
+
+      expect(results).to.have.lengthOf(10);
+
+        results.map(function(obj, index) {
+          index += 1;
+          expect(obj.order).to.be.equal(index);
+        });
 
-      expect(results).to.have.lengthOf(1);
     });
 
-    it('should query by whereIn', async() => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product
-      }).whereIn<string>('name', ['test101', 'test107', 'test106']).get();
-      
-      expect(results).to.have.lengthOf(4);      
+    it('should query with orderBy as string', async () => {
+      const results: Universal[] = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .orderBy<string>('order', 'ASC').get();
+
+      expect(results[1].order).to.be.equal(10);
+
     });
 
-    it('should query by startsWith', async() => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product
-      }).whereStartsWith('name', 'n').get();
-      
-      expect(results[0].name).to.equals('new_testing');
+    it('should query with orderByDESC as long', async () => {
+      const results: Universal[] = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .orderBy<string>('order as long', 'DESC').get();
+
+      expect(results).to.have.lengthOf(10);
+
+      results.reverse().map(function(obj, index) {
+        index += 1;
+        expect(obj.order).to.be.equal(index);
+      });
+
     });
 
-    it('should query by endsWith', async() => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product
-      }).whereEndsWith('name', '7').get();
-      
-      expect(results[0].name).to.equals('test107');
+    it('should query with whereGreaterThanOrEqual', async () => {
+      const results: Universal[] = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .whereGreaterThanOrEqual<number>('order',6, 'name', 'withNesting').get();
+
+      expect(results).to.have.lengthOf(10);
+
+        results.map(function(obj) {
```

Comment:
Assertions are wrong. You test `whereGreaterThanOrEqual` but check whether results contain `name` property set to 'withNesting'. You should verify that every result has property 'order' greater or equal to 6.

Also what are those additional parameters passed to `whereGreaterThanOrEqual` (`...'name', 'withNesting')`?